programs/smart-wallet/src/instructions/buffer_finalize.rs

```diff
@@ -0,0 +1,41 @@
+//! Finalize an [InstructionBuffer] for execution.
+
+use crate::*;
+
+#[derive(Accounts)]
+pub struct FinalizeBuffer<'info> {
+    #[account(mut)]
+    pub buffer: Account<'info, InstructionBuffer>,
+    pub writer: Signer<'info>,
+}
+
+/// Emitted when a [InstructionBuffer] is finalized.
+#[event]
+pub struct FinalizeBufferEvent {
+    /// The buffer.
+    pub buffer: Pubkey,
+    /// The time [InstructionBuffer] is finalized.
+    #[index]
+    pub time: i64,
+}
+
+pub fn handler(ctx: Context<FinalizeBuffer>) -> Result<()> {
+    let buffer = &mut ctx.accounts.buffer;
+    buffer.finalized_at = Clock::get()?.unix_timestamp;
```

Comment:
Delete `finalized_at` and replace with changing the `writer` to `system_program::ID`.